tree/ntuple/v7/test/ntuple_descriptor.cxx

```diff
@@ -235,6 +235,28 @@ TEST(RNTupleDescriptorBuilder, CatchInvalidDescriptors)
    descBuilder.EnsureValidDescriptor();
 }
 
+TEST(RNTupleDescriptor, QualifiedFieldName)
+{
+   auto model = RNTupleModel::Create();
+   model->MakeField<std::int32_t>("ints");
+   model->MakeField<std::vector<float>>("jets");
+   FileRaii fileGuard("test_field_qualified.root");
+   {
+      RNTupleWriter ntuple(std::move(model),
+         std::make_unique<RPageSinkFile>("ntuple", fileGuard.GetPath(), RNTupleWriteOptions()));
+   }
+
+   auto ntuple = RNTupleReader::Open("ntuple", fileGuard.GetPath());
+   const auto &desc = ntuple->GetDescriptor();
+   EXPECT_TRUE(desc.GetQualifiedFieldName(desc.GetFieldZeroId()).empty());
+   auto fldIdInts = desc.FindFieldId("ints");
+   EXPECT_STREQ("ints", desc.GetQualifiedFieldName(fldIdInts).c_str());
+   auto fldIdJets = desc.FindFieldId("jets");
+   auto fldIdInner = desc.FindFieldId("float", fldIdJets);
```

Comment:
That's currently the name choice for the child field of a collection.  Not sure yet if that is a good idea.  The inner field can be named anything (e.g. `@1` would work, too). There can be no name clash because a collection in RNTuple has exactly one sub field.

Reply:
I would find something like `data` more fitting (but I realize this is subjective)